### PR TITLE
Initial code for execution info

### PR DESCRIPTION
**Pull request by sydhds**
* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

File: massa-execution-exports/src/controller_traits.rs [hunk 1]
```diff
@@ -22,7 +22,11 @@ use std::collections::BTreeMap;
 use std::collections::HashMap;
 
 #[cfg(feature = "execution-trace")]
-use crate::types::{AbiTrace, SlotAbiCallStack, Transfer};
+use crate::types_trace_info::AbiTrace;
```
Comment (AurelienFT) [inline massa-execution-exports/src/controller_traits.rs:25]: Can we wrap them in one import like before ?

File: massa-execution-exports/src/lib.rs [hunk 2]
```diff
@@ -72,7 +72,15 @@ pub use types::{
...
 #[cfg(feature = "execution-trace")]
-pub use types::{
-    AbiTrace, SCRuntimeAbiTraceType, SCRuntimeAbiTraceValue, SlotAbiCallStack, Transfer,
-};
+pub use types_trace_info::AbiTrace;
+#[cfg(feature = "execution-trace")]
+pub use types_trace_info::{SCRuntimeAbiTraceType, SCRuntimeAbiTraceValue};
+
+#[cfg(feature = "execution-trace")]
+pub use types_trace_info::SlotAbiCallStack;
```
Comment (AurelienFT) [inline massa-execution-exports/src/lib.rs:84]: Can we wrap them in one `use` ?

File: massa-execution-exports/src/types.rs [hunk 3]
```diff
@@ -361,26 +241,12 @@ pub struct ExecutionOutput {
...
-    /// Effective received amount
-    pub effective_received_amount: Amount,
-    /// operation id
-    pub op_id: OperationId,
-    /// success or not
-    pub succeed: bool,
-    /// Fee
-    pub fee: Amount,
+    /// Deferred credits execution (empty if execution-info feature is NOT enabled)
+    pub deferred_credits_execution: Vec<(Address, Result<Amount, String>)>,
```
Comment (AurelienFT) [inline massa-execution-exports/src/types.rs:245]: Just a question to make sure I understand well, the fields will always exist even if the feature `execution-info` isn't activated but they will be empty ?

File: massa-execution-worker/src/execution_info.rs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    }
+}
+
+pub enum OperationInfo {
+    RollBuy(u64),
+    RollSell(u64),
+}
+
+pub struct ExecutionInfoForSlot {
+    pub(crate) block_producer_reward: Option<(Address, Amount)>,
```
Comment (AurelienFT) [inline massa-execution-worker/src/execution_info.rs:24]: I don't understand why there is less fields in the structure `ExecutionOutput` of `massa-execution-exports` than here. Moreover it's not possible to have common structure defined in `massa-execution-exports` and used in this module ?

File: massa-execution-exports/src/types.rs (hunk 3 above)
Comment (bilboquet) [reply massa-execution-exports/src/types.rs:245]: If the feature is not defined, the field does not exist.
Example here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=31d2d33c65d9f35de77a00435365fbe1

File: massa-execution-exports/src/controller_traits.rs (hunk 1 above)
Comment (sydhds) [reply massa-execution-exports/src/controller_traits.rs]: Done

File: massa-execution-exports/src/lib.rs (hunk 2 above)
Comment (sydhds) [reply massa-execution-exports/src/lib.rs]: Done

File: massa-execution-worker/src/execution_info.rs (hunk 4 above)
Comment (sydhds) [reply massa-execution-worker/src/execution_info.rs:37]: In fact, I forgot to add ExecutionInfoForSlot into ExecutionInfo (which is in ExecutionOutput) but I've just fix this...

**Comment by sydhds** [comment]
ExecuteInfoForSlot has more fields than ExecutionOutput so it makes sense to have both structures as it is defined in this PR. Let me know if this is still an issue?

**Review by AurelienFT — APPROVED** [review]

**Pull request merged**